cli/src/main.rs

```diff
@@ -1,31 +1,84 @@
 #![allow(dead_code)]
-mod cli;
-mod repl;
-mod report;
 
-use crate::cli::{handle_source, Cli};
-use crate::repl::prompt;
-use clap::Parser;
-use context::source::Source;
-use miette::MietteHandlerOpts;
-use std::io;
+use ariadne::{Color, StdoutFmt};
 use std::ops::Deref;
+use std::path::PathBuf;
 use std::process::exit;
 
-fn main() -> io::Result<()> {
+use clap::Parser;
+
+use crate::cli::{Cli, Configuration};
+use crate::repl::repl;
+use crate::runner::run;
+
+mod cli;
+mod display_bytecode;
+mod formatted_diagnostic;
+mod formatted_parse_error;
+mod repl;
+mod runner;
+mod source_importer;
+
+fn main() {
     let cli = Cli::parse();
 
-    miette::set_hook(Box::new(|_| {
-        Box::new(MietteHandlerOpts::new().tab_width(2).build())
-    }))
-    .expect("miette options setup");
+    let config = Configuration::from(cli.clone());
+
+    let working_dir = update_working_dir(cli.working_dir);
 
     if let Some(source) = cli.source {
-        let content = std::fs::read_to_string(&source)?;
-        let name = source.to_string_lossy().deref().to_string();
-        let source = Source::new(&content, &name);
-        exit(handle_source(source) as i32)
+        exit(run(source, working_dir, config) as i32)
+    }
+    repl(config);
+}
+
+fn update_working_dir(working_dir: Option<PathBuf>) -> PathBuf {
```

Comment:
An erroneous working directory should return an error and not `exit(1)` the program.

Reply:
it's just an utility function, it is declared in `main` and is not meant to be used in any other way that its currently used

Reply:
So they are three main issues:
- Just panic if the working directory cannot be changed. If you want to customize it more, it is a sign that you shouldn't use blindly `exit(1)`.
- Having the working directory existing doesn't mean that `set_current_dir()` will success if the directory has restrictive permissions.
- If the user wants to change the directory, they can just change it before invoking the cli. It is a source path, not a working directory path.